cli/docker_validator.py

```diff
@@ -0,0 +1,205 @@
+import argparse
+import csv
+import logging
+import subprocess
+from pathlib import Path
+
+logging.basicConfig(format='%(asctime)s %(message)s', datefmt='%m/%d/%Y %I:%M:%S %p', level='INFO')
+
+container_image = 'eva_pre_submission_validator'
+container_validation_dir = '/opt/vcf_validation'
+container_validation_output_dir = '/opt/vcf_validation/vcf_validation_output'
+
+
+def run_command_with_output(command_description, command, return_process_output=True,
+                            log_error_stream_to_output=False):
+    process_output = ""
+
+    logging.info("Starting process: " + command_description)
+    logging.info("Running command: " + command)
+
+    stdout = subprocess.PIPE
+    # Some utilities output non-error messages to error stream. This is a workaround for that
+    stderr = subprocess.STDOUT if log_error_stream_to_output else subprocess.PIPE
+
+    with subprocess.Popen(command, stdout=stdout, stderr=stderr, bufsize=1, universal_newlines=True,
+                          shell=True) as process:
+        for line in iter(process.stdout.readline, ''):
+            line = str(line).rstrip()
+            logging.info(line)
+            if return_process_output:
+                process_output += line + "\n"
+        if not log_error_stream_to_output:
+            for line in iter(process.stderr.readline, ''):
+                line = str(line).rstrip()
+                logging.error(line)
+    if process.returncode != 0:
+        logging.error(command_description + " failed! Refer to the error messages for details.")
+        raise subprocess.CalledProcessError(process.returncode, process.args)
+    else:
+        logging.info(command_description + " - completed successfully")
+    if return_process_output:
+        return process_output
+
+
+def check_if_file_missing(mapping_file):
+    files_missing = False
+    with open(mapping_file) as open_file:
+        reader = csv.DictReader(open_file, delimiter=',')
+        for row in reader:
+            if not Path(row['vcf']).is_file():
+                files_missing = True
+                logging.error('%s does not exist', row['vcf'])
+
+    return files_missing
+
+
+def verify_docker_is_installed(docker):
+    try:
+        run_command_with_output("check docker is installed and available on the path", f"{docker} --version")
+    except subprocess.CalledProcessError as ex:
+        logging.error(ex)
+        raise RuntimeError(f"Please make sure docker ({docker}) is installed and available on the path")
+
+
+def verify_container_is_running(docker, container_name):
+    container_run_cmd_ouptut = run_command_with_output("check if container is running", f"{docker} ps")
```

Comment:
Nice Catch Sundar. 
Since it was running on my machine without sudo, I forgot that it does not by default. We need to add the user to the docker group.
Not sure, if we want to handle sudo in our code or just let the user know of this restriction.